src/backend/utils/misc/queryenvironment.c

```diff
@@ -878,15 +878,21 @@ static bool _ENR_tuple_operation(Relation catalog_rel, HeapTuple tup, ENRTupleOp
 					CacheInvalidateHeapTuple(catalog_rel, newtup, NULL);
 					break;
 				case ENR_OP_UPDATE:
-					oldtup = lfirst(lc);
-					lfirst(lc) = heap_copytuple(tup);
-					CacheInvalidateHeapTuple(catalog_rel, oldtup, tup);
+					if (lc)
```

Comment:
There is a case in table_variable_xact_nested where lc is NULL causing a crash. I will look into it more.

Reply:
The lc is NULL is related to this https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/232#discussion_r1366272681

Reply:
We should remove the lc is NULL check since the oid change should not affect the logic here.
Unless there is a separate bug you're trying to fix.

I think once the issue in my comment https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/232#discussion_r1366272681 is fixed then you do not need to add the if(lc) check here anymore.